python/lib/sift_py/upload/config.py

```diff
@@ -0,0 +1,149 @@
+from typing import Dict, List, Optional
+
+from pydantic import BaseModel, ConfigDict, model_validator
+from pydantic_core import PydanticCustomError
+
+VALID_TIME_FORMATS = [
+    "TIME_FORMAT_ABSOLUTE_RFC3339",
+    "TIME_FORMAT_ABSOLUTE_DATETIME",
+    "TIME_FORMAT_ABSOLUTE_UNIX_SECONDS",
+    "TIME_FORMAT_ABSOLUTE_UNIX_MILLISECONDS",
+    "TIME_FORMAT_ABSOLUTE_UNIX_MICROSECONDS",
+    "TIME_FORMAT_ABSOLUTE_UNIX_NANOSECONDS",
+    "TIME_FORMAT_RELATIVE_NANOSECONDS",
+    "TIME_FORMAT_RELATIVE_MICROSECONDS",
+    "TIME_FORMAT_RELATIVE_MILLISECONDS",
+    "TIME_FORMAT_RELATIVE_SECONDS",
+    "TIME_FORMAT_RELATIVE_MINUTES",
+    "TIME_FORMAT_RELATIVE_HOURS",
+]
+
+
+VALID_DATA_TYPES = [
+    "CHANNEL_DATA_TYPE_DOUBLE",
+    "CHANNEL_DATA_TYPE_FLOAT",
+    "CHANNEL_DATA_TYPE_STRING",
+    "CHANNEL_DATA_TYPE_BOOL",
+    "CHANNEL_DATA_TYPE_INT_32",
+    "CHANNEL_DATA_TYPE_INT_64",
+    "CHANNEL_DATA_TYPE_UINT_32",
+    "CHANNEL_DATA_TYPE_UINT_6",
+    "CHANNEL_DATA_TYPE_ENUM",
+    "CHANNEL_DATA_TYPE_BIT_FIELD",
+]
+
+
+class _BaseModel(BaseModel):
+    model_config = ConfigDict(extra="forbid")
+
+
+class EnumType(_BaseModel):
```

Comment:
We have `ChannelEnumType` over in `lib/sift_py/ingestion/channel.py`.

Reply:
I think the best path forward is to inherit from both BaseModel and ChannelEnumType, that way the ChannelEnumType class is the source of truth and we get the validation from BaseModel.